pre-commit/strip.sh

```diff
@@ -0,0 +1,14 @@
+#!/bin/bash
+
+# strip all TARGET string what matches PATTERN
+# sh strip.sh abcb b -> ab
+
+if [ $# -ne 2 ]; then
+  echo "2 arguments are required"
+  echo "1st arg is source string, 2nd arg is pattern string"
+  exit 1
+fi
+
+SOURCE="$1"
+PATTERN="$2"
+echo "${SOURCE//"$PATTERN"}" # replace all. `//` means global
```

Comment:
This really isn't what you want based on how the script is actually called.  What you _want_ is to remove a few directory components (a "prefix") from the path.  So really...
```suggestion
echo "${SOURCE#"$PATTERN"}" # remove pattern from beginning of source
```

Reply:
the SOURE can have multiple paths like `webui/react/src/pages/ProjectDetails.settings.ts webui/react/src/pages/ProjectDetails.tsx`, so I'll leave it with `//`

Reply:
Then the args are backwards.  You should pass the pattern in first, and then treat additional files as multiple arguments.

Reply:
oh that's true. I'll change it